src/machine-c-version-template.h

```diff
@@ -0,0 +1,35 @@
+// Copyright 2023 Cartesi Pte. Ltd.
```

Comment:
Since we have a templates folder, we should we put this file there.
`machine-c-version-template.h` -> `machine-c-version.h.template`